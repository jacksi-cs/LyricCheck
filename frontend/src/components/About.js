import axios from "axios";
import React, { useState, useEffect } from "react";
import christopher_img from "./images/christopher.png";
import stacy_img from "./images/stacy.png";
import kenji_img from "./images/kenji.jpg";
import jack_img from "./images/jack.png";
import hassan_img from "./images/hassan.jpg";
import aws from "./images/about/aws.png";
import docker from "./images/about/docker.png"
import elastic_beanstalk from "./images/about/elastic_beanstalk.jpg";
import eslint from "./images/about/eslint.png";
import flask_restless from "./images/about/flask_restless.png";
import gitlab from "./images/about/gitlab.jpg";
import gitlablogo from  "./images/about/gitlablogo.jpg"
import namecheap from "./images/about/namecheap.png";
import postgresql from "./images/about/postgresql.png";
import postman from "./images/about/postman.jpg";
import prettier from "./images/about/prettier.png";
import react512 from "./images/about/react512.png";
import lastfm from "./images/about/last.fm.png";
import mui from "./images/about/mui.png";
import musixmatch from "./images/about/musixmatch.png";
import perspective from "./images/about/perspective.png";
import RESTcountries from "./images/about/RESTcountries.png";
import spotify from "./images/about/spotify.png";

import Container from "react-bootstrap/Container";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";
import Card from "react-bootstrap/Card";
import Image from "react-bootstrap/Image";

import Divider from "@mui/material/Divider"

export const scrape = async (url) => {
  return await axios.get(url);
};

const team = [
  {
    name: "Hassan Khan",
    email: "hakhan0301@utexas.edu",
    username: "hakhan0301",
    image: hassan_img,
    bio: "I'm a Junior in CS at UT Austin. I moved to Texas when I was 5 and have been here since. I like playing Minecraft.",
    responsibilities: "Backend | Phase I Leader",
    commits: 0,
    issues: 0,
    tests: 12,
  },
  {
    name: "Stacy Jong",
    email: "stacyjong817@gmail.com",
    username: "stacyvjong",
    image: stacy_img,
    bio: "I am a third-year CS major at UT Austin, and am from Seattle, WA. In my free-time I enjoy playing board games with friends, playing tennis, and hiking.",
    responsibilities: "Backend | Phase IV Leader",
    commits: 0,
    issues: 0,
    tests: 6,
  },
  {
    name: "Jack Si",
    email: "jacksi@utexas.edu",
    username: "jacksi1",
    image: jack_img,
    bio: "I'm an avid music listener who plays basketball and golf whenever possible. Currently in my third year, I'm a CS major at UT Austin.",
    responsibilities: "Frontend | Phase III Leader",
    commits: 0,
    issues: 0,
    tests: 26,
  },
  {
    name: "Kenji Nakachi",
    email: "kenjinakachi@utexas.edu",
    username: "kenjinakachi",
    image: kenji_img,
    bio: "I moved to Texas when I was 8 and have lived in Kyle ever since. I'm a third year CS major at UT Austin, and enjoy playing basketball and video games in my free time.",
    responsibilities: "Backend",
    commits: 0,
    issues: 0,
    tests: 0,
  },
  {
    name: "Christopher Carrasco",
    email: "chris.carrasco@att.net",
    username: "guymanpersonboy",
    image: christopher_img,
    bio: "I'm a third year CS major at UT Austin. I'm from Midland, TX and enjoy making music, gaming, watching anime, and podcasting with friends.",
    responsibilities: "Frontend | Phase II Leader",
    commits: 0,
    issues: 0,
    tests: 15,
  },
];

function About() {
  const [teamInfo, setTeamInfo] = useState([]);
  const [totalCommits, setTotalCommits] = useState([]);
  const [totalIssues, setTotalIssues] = useState([]);
  const [totalTests, setTotalTests] = useState([]);

  const getCommits = async () => {
    let totalCommits = 0;

    const commitInfo = await scrape(
      "https://gitlab.com/api/v4/projects/33830147/repository/contributors"
    );

    commitInfo.data.forEach((element) => {
      team.forEach((member) => {
        if (
          member.name.includes(element.name) ||
          member.email === element.email
        ) {
          member.commits += element.commits;
        }
      });
      totalCommits += element.commits;
    });
    setTotalCommits(totalCommits);
  };

  const getIssues = async () => {
    let totalIssues = 0;

    const issueList = await scrape(
      "https://gitlab.com/api/v4/projects/33830147/issues"
    );

    issueList.data.forEach((element) => {
      let assigned = 0;
      element.assignees.forEach((a) => {
        team.forEach((member) => {
          if (member.username === a.username) {
            member.issues += 1;
            assigned += 1;
          }
        });
      });

      team.forEach((member) => {
        if (!assigned && element.assignee != null && member.username === element.assignee.username) {
          member.issues += 1;
        }
      });
      totalIssues += 1;
    });
    setTotalIssues(totalIssues);
    setTeamInfo(team);
  };

  const getTests = async () => {
    let totalTests = 0;
    team.forEach((member) => {
      totalTests += member.tests
    });
    setTotalTests(totalTests);
  };

  useEffect(() => {
    const getData = async () => {
      getCommits();
      getIssues();
      getTests();
    };
    getData();
  }, []);

  return (
    <Container style={{ paddingBottom: '25px' }}>
      <Row style={{ paddingTop: '50px', paddingBottom: '25px' }}>
        <h1 className="text-center">About Us</h1>
        <h2>What is LyricCheck?</h2>
        <p>
          LyricCheck is a website that consolidates data on top artists (monthly
          listeners), charting songs, artist bios, song lyrics, and profanity
          scores. Using Perspective's NLP model to grade song lyrics, our project
          will also provide insight on the profanity preferences over time, based
          on location, and based on artists. It will additionally have a Spotify
          playlist analyzer that will grade the profanity level of Spotify
          playlists, allowing for a bulk solution to children-safe songs. There
          are sites that tackle one or even part of one responsibility that we
          hope our project to accomplish, but there are severe patches of
          information missing from each. By also making use of a profanity grade,
          our project can provide valuable insight into the worrying NSFW trend of
          modern day pop culture.
        </p>

        <h2>What has been found?</h2>
        <p>
          Of the 207 country instances, Palau has the highest profanity
          average while Denmark has the lowest. Of the 1239 song instances
          we have, "Under Ground Kings" by Drake has the highest profanity average
          while "Dreams" by The Cranberries has the lowest profanity average. Hip hop
          and Rap genres tend to have high profanity averages. Of the 776 artist
          instances, A$AP Rocky has the highest profanity average while Bastille has
          the lowest profanity average.
        </p>
      </Row>

      <Divider></Divider>

      <Row style={{ paddingTop: '50px', paddingBottom: '25px' }}>
        <h1 className="text-center">Our Team</h1>
        <Row>
          {teamInfo.map((member) => (
            <Col>
              <Card>
                <Card.Img variant="top" src={member.image} className="card-img" />
                <Card.Body>
                  <Card.Title>
                    {member.name} - {member.responsibilities}
                  </Card.Title>
                  <Card.Text>
                    Responsibilities: {member.responsibilities}
                    <br />
                    Bio: {member.bio}
                    <br />
                  </Card.Text>
                </Card.Body>
                <Card.Footer>
                  Commits: {member.commits}
                  <br />
                  Issues: {member.issues}
                  <br />
                  Unit Tests: {member.tests}
                </Card.Footer>
              </Card>
            </Col>
          ))}
        </Row>
        <Row>
          <Col>
            <h5 className="text-center">Total number of commits: {totalCommits}</h5>
          </Col>
          <Col>
            <h5 className="text-center">Total number of issues: {totalIssues}</h5>
          </Col>
          <Col>
            <h5 className="text-center">Total number of tests: {totalTests}</h5>
          </Col>
        </Row>
      </Row>

      <Divider></Divider>

      <Row style={{ paddingTop: '50px', paddingBottom: '25px' }}>
        <Row>
          <Col align="center"><h2>Data Sources</h2></Col>
        </Row>

        <Row>
          <Col>
            <div onClick = {() => (window.location.href = "https://restcountries.com/")}>
              <Card>
                <Card.Img variant="top" src={RESTcountries} className="tool-img" />
                <Card.Body>
                  <Card.Title>
                    REST Countries
                  </Card.Title>
                  <Card.Text>
                    Was scraped using node-fetch. We got out a list of all the
                    countries.
                  </Card.Text>
                </Card.Body>
              </Card>
            </div>
          </Col>
          <Col>
            <div onClick = {() => (window.location.href = "https://www.last.fm/api/intro")}>
              <Card>
                <Card.Img variant="top" src={lastfm} className="tool-img" />
                <Card.Body>
                  <Card.Title>
                    last.fm
                  </Card.Title>
                  <Card.Text>
                    Was scraped using node-fetch. We input a country, and got out top
                    songs and artists. We also input song and artist names to get data
                    about those.
                  </Card.Text>
                </Card.Body>
              </Card>
            </div>
          </Col>
          <Col>
            <div onClick = {() => (window.location.href = "https://developer.musixmatch.com/")}>
              <Card>
                <Card.Img variant="top" src={musixmatch} className="tool-img" />
                <Card.Body>
                  <Card.Title>
                    Musicxmatch
                  </Card.Title>
                  <Card.Text>
                    Was scraped using node-fetch. We input a song name, and got out the
                    lyrics.
                  </Card.Text>
                </Card.Body>
              </Card>
            </div>
          </Col>
          <Col lg md={3} sm={4} xs={4}>
            <div onClick = {() => (window.location.href = "https://perspectiveapi.com/")}>
              <Card>
                <Card.Img variant="top" src={perspective} className="tool-img" />
                <Card.Body>
                  <Card.Title>
                    Perspective
                  </Card.Title>
                  <Card.Text>
                    Was scraped using node-fetch. We gave lyrics and got the profanity
                    score.
                  </Card.Text>
                </Card.Body>
              </Card>
            </div>
          </Col>
          <Col lg md={3} sm={4} xs={4}>
            <div onClick = {() => (window.location.href = "https://developer.spotify.com/documentation/web-api/")}>
              <Card>
                <Card.Img variant="top" src={spotify} className="tool-img" />
                <Card.Body>
                  <Card.Title>
                    Spotify
                  </Card.Title>
                  <Card.Text>
                    Was scraped using node-fetch. We initially used this to get song and
                    artist info, but we later only used it for getting media links.
                  </Card.Text>
                </Card.Body>
              </Card>
            </div>
          </Col>
        </Row>

        <Row>
          <Col align="center"><h2>Tools</h2></Col>
        </Row>

        <Row>
          <Col>
            <div onClick = {() => (window.location.href = "https://reactjs.org")}>
              <Card>
                <Card.Img variant="top" src={react512} className="tool-img" />
                <Card.Body>
                  <Card.Title>
                    React
                  </Card.Title>
                  <Card.Text>
                    A Javascript library used for web application development.
                  </Card.Text>
                </Card.Body>
              </Card>
            </div>
          </Col>
          <Col>
            <div onClick = {() => (window.location.href = "https://www.postman.com")}>
              <Card>
                <Card.Img variant="top" src={postman} className="tool-img" />
                <Card.Body>
                  <Card.Title>
                    Postman
                  </Card.Title>
                  <Card.Text>
                    An API platform used for building APIs.
                  </Card.Text>
                </Card.Body>
              </Card>
            </div>
          </Col>
          <Col>
            <div onClick = {() => (window.location.href = "https://aws.amazon.com")}>
              <Card>
                <Card.Img variant="top" src={aws} className="tool-img" />
                <Card.Body>
                  <Card.Title>
                    AWS
                  </Card.Title>
                  <Card.Text>
                    A cloud platform used for hosting the React app.
                  </Card.Text>
                </Card.Body>
              </Card>
            </div>
          </Col>
          <Col lg md={4} sm={5} xs={4}>
            <div onClick = {() => (window.location.href = "https://aws.amazon.com/elasticbeanstalk/")}>
              <Card>
                <Card.Img variant="top" src={elastic_beanstalk} className="tool-img" />
                <Card.Body>
                  <Card.Title>
                    Elastic Beanstalk
                  </Card.Title>
                  <Card.Text>
                    Cloud tool used to host our API.
                  </Card.Text>
                </Card.Body>
              </Card>
            </div>
          </Col>
          <Col lg md={4} sm={4} xs={4}>
            <div onClick = {() => (window.location.href = "https://www.postgresql.org")}>
              <Card>
                <Card.Img variant="top" src={postgresql} className="tool-img" />
                <Card.Body>
                  <Card.Title>
                    PostgreSQL
                  </Card.Title>
                  <Card.Text>
                    Database used to store and search model data.
                  </Card.Text>
                </Card.Body>
              </Card>
            </div>
          </Col>
        </Row>

        <Row>
          <Col>
            <div onClick = {() => (window.location.href = "https://flask-restless-ng.readthedocs.io/en/latest/")}>
              <Card>
                <Card.Img variant="top" src={flask_restless} className="tool-img" />
                <Card.Body>
                  <Card.Title>
                    Flask-Restless-NG
                  </Card.Title>
                  <Card.Text>
                    Simple generation of ReSTful APIs for database models defined using SQLAlchemy (or Flask-SQLAlchemy).
                  </Card.Text>
                </Card.Body>
              </Card>
            </div>
          </Col>
          <Col>
            <div onClick = {() => (window.location.href = "https://www.docker.com")}>
              <Card>
                <Card.Img variant="top" src={docker} className="tool-img" />
                <Card.Body>
                  <Card.Title>
                    Docker
                  </Card.Title>
                  <Card.Text>
                    Uses OS-level virtualization to deliver software in packages called containers.
                  </Card.Text>
                </Card.Body>
              </Card>
            </div>
          </Col>
          <Col>
            <div onClick = {() => (window.location.href = "https://gitlab.com")}>
              <Card>
                <Card.Img variant="top" src={gitlab} className="tool-img" />
                <Card.Body>
                  <Card.Title>
                    Gitlab
                  </Card.Title>
                  <Card.Text>
                    An open source code repository and DevOps platform.
                  </Card.Text>
                </Card.Body>
              </Card>
            </div>
          </Col>
          <Col lg md={3} sm={5} xs={4}>
            <div onClick = {() => (window.location.href = "https://www.namecheap.com")}>
              <Card>
                <Card.Img variant="top" src={namecheap} className="tool-img" />
                <Card.Body>
                  <Card.Title>
                    Namecheap
                  </Card.Title>
                  <Card.Text>
                    A domain name registrar.
                  </Card.Text>
                </Card.Body>
              </Card>
            </div>
          </Col>
          <Col lg md={4} sm={4} xs={4}>
            <div onClick = {() => (window.location.href = "https://eslint.org")}>
              <Card>
                <Card.Img variant="top" src={eslint} className="tool-img" />
                <Card.Body>
                  <Card.Title>
                    ESLint
                  </Card.Title>
                  <Card.Text>
                    An open source JavaScript linting tool.
                  </Card.Text>
                </Card.Body>
              </Card>
            </div>
          </Col>
        </Row>  

        <Row>
          <Col xl={3} lg={3} md={4} sm={4} xs={4}>
            <div onClick = {() => (window.location.href = "https://prettier.io")}>
              <Card>
                <Card.Img variant="top" src={prettier} className="tool-img" />
                <Card.Body>
                  <Card.Title>
                    Prettier
                  </Card.Title>
                  <Card.Text>
                    A code formatter.
                  </Card.Text>
                </Card.Body>
              </Card>
            </div>
          </Col>
          <Col xl={3} lg={3} md={4} sm={4} xs={4}>
            <div onClick = {() => (window.location.href = "https://mui.com")}>
              <Card>
                <Card.Img variant="top" src={mui} className="tool-img" />
                <Card.Body>
                  <Card.Title>
                    Material UI
                  </Card.Title>
                  <Card.Text>
                    A comprehensive suite of UI tools.
                  </Card.Text>
                </Card.Body>
              </Card>
            </div>
          </Col>
        </Row>
      </Row>

      <Divider></Divider>

      <Row style={{ paddingTop: '50px', paddingBottom: '10px' }}>
        <Row>
          <Col align="center"><h2>Our Repository</h2></Col>
          <Col align="center"><h2>API Documentation</h2></Col>
        </Row>
        
        <Row>
          <Col>
            <Row>
              <Col align="center" xs={5}>
                <Image src={gitlablogo} alt="tool image" className="tool-img"></Image>
              </Col>
              <Col>
                <li><a href="https://gitlab.com/kenjinakachi/lyriccheck">GitLab Repository</a></li>
                <p>
                  GitLab respository containing our source code.
                </p>
              </Col>
            </Row>
          </Col>
          <Col>
            <Row>
              <Col align="center" xs={5}>
                <Image src={postman} alt="tool image" style={{height: 92}}></Image>
              </Col>
              <Col>
                <li><a href="https://documenter.getpostman.com/view/19683184/UVksKDAj">Postman API</a></li>
                <p>Postman documentation of LyricCheck's API</p>
              </Col>
            </Row>
          </Col>
        </Row>
      </Row>
    </Container>
  );
}

export default About;
