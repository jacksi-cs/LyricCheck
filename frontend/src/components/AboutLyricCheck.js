import Container from "react-bootstrap/Container";
import perspective from "./images/about-lyriccheck/perspective-ex.png";

function AboutLyricCheck() {
  return (
    <>
      <Container style={{ paddingTop: '50px', paddingBottom: '50px' }}>
        <h1 class="text-center">Profanity Metrics</h1>
        <h2>How are songs rated based on profanity?</h2>
        <img src={perspective} alt="perspective example" width="70%" height="70%"></img>
        <p>
          We pull the profanity metrics from <a href="https://perspectiveapi.com/">Perspective API</a>, 
          which evaluates song lyrics on several categories. We focus on the following categories:&nbsp;
          <b>Severe Toxicity</b>, <b>Profanity</b>, <b>Sexually Explicit</b>, and <b>Identity Attack</b>. We also include an average
          profanity score of each song based on these four categories, which is what we use to rank songs, 
          artists, and countries based on their profanity score.. Below is a description of the four
          categories, taken directly from the Perspective API site.
        </p>
        <h4>Severe Toxicity</h4>
        <p>
          A very hateful, aggressive, disrespectful comment or otherwise very likely to make a user leave 
          a discussion or give up on sharing their perspective. This attribute is much less sensitive to 
          more mild forms of toxicity, such as comments that include positive uses of curse words.
        </p>
        <h4>Profanity</h4>
        <p>
          Swear words, curse words, or other obscene or profane language.
        </p>
        <h4>Sexually Explicit</h4>
        <p>
          Contains references to sexual acts, body parts, or other lewd content.
        </p>
        <h4>Identity Attack</h4>
        <p>
          Negative or hateful comments targeting someone because of their identity.
        </p>  
      </Container>

      <Container style={{ paddingBottom: '75px' }}>
        <h1 class="text-center">Views</h1>
        <h2>Where do view counts come from?</h2>
        <p>
          The view count for all songs are provided by Last.fm.
          For more information, please refer to&nbsp;
          <a href="https://www.last.fm/about">last.fm/about</a>.
        </p>
      </Container>
    </>
  );
}

export default AboutLyricCheck;