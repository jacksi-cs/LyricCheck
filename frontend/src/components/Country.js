import { useParams } from "react-router-dom";
import Container from "react-bootstrap/Container";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";
import { useEffect, useState } from "react";
import { getCountry } from "../lib/api-helpers";
import { Link } from "react-router-dom";
import Tabs from "react-bootstrap/Tabs";
import Tab from "react-bootstrap/Tab";


function Country() {
  const [loading, setLoading] = useState(true);
  const [country, setCountry] = useState(null);

  const { id } = useParams();

  useEffect(() => {
    getCountry(id).then(setCountry).then(() => setLoading(false));
  }, [id]);

  if (loading) {
    return <h1>Loading....</h1>;
  }
  if (!country) {
    return <>
      <h1>Country not available</h1>;
    </>;
  }

  console.log(country)

  const {
    continents,
    country_code,
    flags_svg,
    languages,
    name_common,
    name_official,
    profanity_identity_attack,
    profanity_profanity,
    profanity_sexually_explicit,
    profanity_toxicity,
    // region,
    subregion,
    top_artists_ids,
    top_artists_names,
    top_tracks_ids,
    top_tracks_names
  } = country;

  const profanityAverage = profanity_identity_attack ? (
    (profanity_identity_attack + profanity_profanity + profanity_sexually_explicit + profanity_toxicity) / 4
  ) : 0;

  return (
    <Container fluid>
      <header>
        <div
            className='p-5 text-center bg-image'
            style={{ backgroundImage: `url(${flags_svg}), url("https://images.pexels.com/photos/2117938/pexels-photo-2117938.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=2")`, height: 300, backgroundRepeat: "no-repeat, repeat-x", backgroundSize: "contain, contain", backgroundPosition: "right, left"}}
        >
          <div className='d-flex justify-content-right align-items-right h-100'>
            <div className='text-white' textAlign = "left">
              <h1 class="text-center">{name_common}</h1>
              <h4 class="text-center">Official Name: {name_official}</h4>
              <h5 class="text-center">Country Code: {country_code}</h5>
              <h5 class="text-center">Located in {subregion}</h5>
              <h5 class="text-center">Continent: {continents[0]} </h5>
            </div>
          </div>
        </div>
      </header>

      <Row className="m-2">
        <iframe
          title={name_common}
          height="450"
          style={{ border: 0 }}
          loading="lazy"
          allowFullScreen
          src={`https://www.google.com/maps/embed/v1/place?key=AIzaSyAy_1LtpV0tyDKPi9-COEdnxj3SDUfy3yM
                &q=${name_common}`}
        />
      </Row>

      <Row className="m-2">
        {profanity_identity_attack ? (
          <Col md="3">
            <h3>Profanity score: </h3>
            <ul>
              <li>Average: {profanityAverage}</li>
              <li>Identity Attack: {profanity_identity_attack}</li>
              <li>Profanity: {profanity_profanity}</li>
              <li>Toxicity: {profanity_toxicity}</li>
              <li>Sexually Explicit: {profanity_sexually_explicit}</li>
            </ul>
          </Col>
        ) : (
          null
        )}

        {languages.length > 0 ? (
          <Col md="3">
            <h3>Languages</h3>
              <ul>
                {Object.values(languages).map((language) => (
                  <li>{language}</li>
                ))}
              </ul>
          </Col>
        ) : (
          null
        )}

        {top_tracks_names.length > 0  ? (
          <Col md="3">
            <h3>Top Songs</h3>
            <ul>
              {top_tracks_names.map((trackName, index) => (
                top_tracks_ids[index]
                  ? <li><Link to={`/song/${top_tracks_ids[index]}`}>{trackName}</Link></li>
                  : <li>{trackName}</li>
              ))}
            </ul>
          </Col>
        ) : (
          null
        )}

        {top_artists_names.length > 0 ? (
          <Col md="3">
            <h3>Top Artists</h3>
            <ul>
              {top_artists_names.map((artistName, index) => (
                top_artists_names[index]
                  ? <li><Link to={`/song/${top_artists_ids[index]}`}>{artistName}</Link></li>
                  : <li>{artistName}</li>
              ))}
            </ul>
          </Col>
        ) : (
          null
        )}
      </Row>
    </Container>
  );
}

export default Country;
