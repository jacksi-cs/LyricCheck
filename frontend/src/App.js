import { Routes, Route } from "react-router-dom";
import { Home } from "./components/Home";
import { Navigation } from "./components/Navigation";
import { FooterContainer } from "./components/Footer";
import About from "./components/About";
import AboutLyricCheck from "./components/AboutLyricCheck";
import OurD3Visuals from "./components/OurD3Visuals";
import TheirD3Visuals from "./components/TheirD3Visuals";

import "./App.css";
import Song from "./components/Song";
import Artist from "./components/Artist";
import Country from "./components/Country";
import SongRankings from "./components/ranking-pages/SongRankings";
import ArtistRankings from "./components/ranking-pages/ArtistRankings";
import CountryRankings from "./components/ranking-pages/CountryRankings";
import Search from "./components/Search"

import "bootstrap/dist/css/bootstrap.min.css";

function App() {
  return (
    <>
      <Navigation />
      <Routes>
        <Route path="/" element={<Home />}></Route>
        <Route path="about" element={<About />}></Route>
        <Route path="song" element={<SongRankings />}></Route>
        <Route path="artist" element={<ArtistRankings />}></Route>
        <Route path="country" element={<CountryRankings />}></Route>

        <Route path="aboutlyriccheck" element={<AboutLyricCheck />}></Route>
        <Route path="ourd3visuals" element={<OurD3Visuals />}></Route>
        <Route path="theird3visuals" element={<TheirD3Visuals />}></Route>

        <Route path="search" element={<Search />}></Route>

        <Route path="song/:id" element={<Song />}></Route>
        <Route path="country/:id" element={<Country />}></Route>
        <Route path="artist/:id" element={<Artist />}></Route>
      </Routes>
      <FooterContainer />
    </>
  );
}

export default App;
