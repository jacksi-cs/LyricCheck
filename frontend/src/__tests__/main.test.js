import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import App from "../App";
// react-bootstrap
import Nav from "react-bootstrap/Nav";
import Navbar from "react-bootstrap/Navbar";
// components
import { Home } from "../components/Home";
import { Navigation } from "../components/Navigation";
import About from "../components/About";
import Song from "../components/Song";
import Artist from "../components/Artist";
import Country from "../components/Country";
import SongRankings from "../components/ranking-pages/SongRankings";
import ArtistRankings from "../components/ranking-pages/ArtistRankings";
import CountryRankings from "../components/ranking-pages/CountryRankings";
import CountryTable from '../components/ranking-pages/CountryTable';
import ArtistTable from '../components/ranking-pages/ArtistTable';
import SongTable from '../components/ranking-pages/SongTable';
import { FooterContainer } from '../components/Footer';
import Footer from '../components/footer/footer-index'
import AboutLyricCheck from '../components/AboutLyricCheck';

configure({ adapter: new Adapter() })

// tests 1-19
describe("rendering components + snapshots", () => {

  // test 1
  test("renders App component without crashing", () => {
    shallow(<App />);
  });

  // test 2
  test("renders all Home component Carousel.Caption without crashing", () => {
    const wrapper = shallow(<Home />);

    const header = (<h1>Welcome to LyricCheck!</h1>);
    const text = (
      <p>
        We consolidate data on top artists, charting songs, and country
        data to analyze the profanity preferences through individual,
        community, and global scopes.
      </p>
    );

    expect(wrapper.contains(header)).toEqual(true);
    expect(wrapper.contains(text)).toEqual(true);

    const headerSlide4 = (<h1>See how countries prefer their profanity in songs!</h1>);
    const textSlide4 = (
      <p>
        Using Perspective's state-of-the-art NLP toxicity models, see how
        a country's profanity tolerance and acceptance changes overtime.
        You can also see the tolerance of profanity around the globe.
      </p>
    );
    expect(wrapper.contains(headerSlide4)).toEqual(true);
    expect(wrapper.contains(textSlide4)).toEqual(true);

    expect(wrapper).toMatchSnapshot()
  })

  // test 3
  test("renders Navigation navbar without crashing", () => {
    const wrapper = shallow(<Navigation />);

    expect(wrapper.contains(<Navbar.Brand href="/">LyricCheck</Navbar.Brand>)).toEqual(true);
    expect(wrapper.contains(<Nav className="me-auto">
      <Nav.Link href="/song">Songs</Nav.Link>
      <Nav.Link href="/artist">Artists</Nav.Link>
      <Nav.Link href="/country">Countries</Nav.Link>
      <Nav.Link href="/about">About Us</Nav.Link>
      <Nav.Link href="/search">Search</Nav.Link>
    </Nav>)).toEqual(true);
    expect(wrapper).toMatchSnapshot()
  })

  // test 4
  test("renders About component without crashing", () => {
    const wrapper = shallow(<About />);
    const header = (<h1 className="text-center">About Us</h1>);
    expect(wrapper.contains(header)).toEqual(true);
    expect(wrapper).toMatchSnapshot()
  })
  // test 5
  test("renders Song component without crashing", () => {
    const wrapper = shallow(<Song />);
    expect(wrapper).toMatchSnapshot()
  })
  // test 6
  test("renders Artist component without crashing", () => {
    const wrapper = shallow(<Artist />);
    expect(wrapper).toMatchSnapshot()
  })
  // test 7
  test("renders Country component without crashing", () => {
    const wrapper = shallow(<Country />);
    expect(wrapper).toMatchSnapshot()
  })
  // test 8
  test("renders SongRankings component without crashing", () => {
    const wrapper = shallow(<SongRankings />);
    expect(wrapper).toMatchSnapshot()
  })
  // test 9
  test("renders ArtistRankings component without crashing", () => {
    const wrapper = shallow(<ArtistRankings />);
    expect(wrapper).toMatchSnapshot()
  })
  // test 10
  test("renders CountryRankings component without crashing", () => {
    const wrapper = shallow(<CountryRankings />);
    expect(wrapper).toMatchSnapshot();
  })
  // test 11
  test("renders AboutLyricCheck without crashing", () => {
    const wrapper = shallow(<AboutLyricCheck />);
    const headerProfanity = (<h1 class="text-center">Profanity Metrics</h1>);
    const headerViews = (<h1 class="text-center">Views</h1>);
    expect(wrapper.contains(headerProfanity)).toEqual(true);
    expect(wrapper.contains(headerViews)).toEqual(true);
    expect(wrapper).toMatchSnapshot();
  })
  // test 12
  test("renders Footer without crashing", () => {
    const wrapper = shallow(<FooterContainer />);
    const title = (<Footer.Title>10 am Group 7</Footer.Title>);
    const text = (<Footer.Text>Â© Copyright 2022 LyricCheck</Footer.Text>);
    expect(wrapper.contains(title)).toEqual(true);
    expect(wrapper.contains(text)).toEqual(true);
    expect(wrapper).toMatchSnapshot();
  })
  // test 13
  test("renders filtered countries without crashing", () => {
    const wrapper = shallow(<CountryTable filters={{ genres: ['rock'] }} />);
    expect(wrapper).toMatchSnapshot();
  })
  // test 14
  test("renders filtered artist without crashing", () => {
    const wrapper = shallow(<ArtistTable filters={{ genres: ['rock'] }} />);
    expect(wrapper).toMatchSnapshot();
  })
  // test 15
  test("renders filtered songs without crashing", () => {
    const wrapper = shallow(<SongTable filters={{ genres: ['rock'] }} />);
    expect(wrapper).toMatchSnapshot();
  })
  // test 16
  test("renders searched countries without crashing", () => {
    const wrapper = shallow(<CountryTable filters={{ search: 'taylor swift' }} />);
    expect(wrapper).toMatchSnapshot();
  })
  // test 17
  test("renders searched artist without crashing", () => {
    const wrapper = shallow(<ArtistTable filters={{ search: 'taylor swift' }} />);
    expect(wrapper).toMatchSnapshot();
  })
  // test 18
  test("renders searched songs without crashing", () => {
    const wrapper = shallow(<SongTable filters={{ search: 'taylor swift' }} />);
    expect(wrapper).toMatchSnapshot();
  })
  // test 19
  test("renders sorted countries without crashing", () => {
    const wrapper = shallow(<CountryTable filters={{ sort: 'genre' }} />);
    expect(wrapper).toMatchSnapshot();
  })
  // test 20
  test("renders sorted artist without crashing", () => {
    const wrapper = shallow(<ArtistTable filters={{ sort: 'genre' }} />);
    expect(wrapper).toMatchSnapshot();
  })
  // test 21
  test("renders sorted songs without crashing", () => {
    const wrapper = shallow(<SongTable filters={{ sort: 'genre' }} />);
    expect(wrapper).toMatchSnapshot();
  })

})
