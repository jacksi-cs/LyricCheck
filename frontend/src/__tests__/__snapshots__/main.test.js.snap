// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering components + snapshots renders About component without crashing 1`] = `ShallowWrapper {}`;

exports[`rendering components + snapshots renders AboutLyricCheck without crashing 1`] = `ShallowWrapper {}`;

exports[`rendering components + snapshots renders Artist component without crashing 1`] = `ShallowWrapper {}`;

exports[`rendering components + snapshots renders ArtistRankings component without crashing 1`] = `ShallowWrapper {}`;

exports[`rendering components + snapshots renders Country component without crashing 1`] = `ShallowWrapper {}`;

exports[`rendering components + snapshots renders CountryRankings component without crashing 1`] = `ShallowWrapper {}`;

exports[`rendering components + snapshots renders Footer without crashing 1`] = `ShallowWrapper {}`;

exports[`rendering components + snapshots renders Navigation navbar without crashing 1`] = `ShallowWrapper {}`;

exports[`rendering components + snapshots renders Song component without crashing 1`] = `ShallowWrapper {}`;

exports[`rendering components + snapshots renders SongRankings component without crashing 1`] = `ShallowWrapper {}`;

exports[`rendering components + snapshots renders all Home component Carousel.Caption without crashing 1`] = `ShallowWrapper {}`;

exports[`rendering components + snapshots renders filtered artist without crashing 1`] = `ShallowWrapper {}`;

exports[`rendering components + snapshots renders filtered countries without crashing 1`] = `ShallowWrapper {}`;

exports[`rendering components + snapshots renders filtered songs without crashing 1`] = `ShallowWrapper {}`;

exports[`rendering components + snapshots renders searched artist without crashing 1`] = `ShallowWrapper {}`;

exports[`rendering components + snapshots renders searched countries without crashing 1`] = `ShallowWrapper {}`;

exports[`rendering components + snapshots renders searched songs without crashing 1`] = `ShallowWrapper {}`;

exports[`rendering components + snapshots renders sorted artist without crashing 1`] = `ShallowWrapper {}`;

exports[`rendering components + snapshots renders sorted countries without crashing 1`] = `ShallowWrapper {}`;

exports[`rendering components + snapshots renders sorted songs without crashing 1`] = `ShallowWrapper {}`;
